### PR TITLE
[Azure.Core] Test Environment Variables

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to include additional environment variables in the `TestEnvironment` to allow for cross-cloud management plane operations,
such as those used by the Azure Messaging client libraries.  This is the first step in enabling cross-cloud support for Live testing in the Event Hubs and Service Bus client libraries, both T1 and T2.

The test resource creation script has been updated to populate these additional variables using the current Azure environment context.   Orthogonal, but included are some tweaks to allow local runs of the `TestResources.ps1` script using the `-WhatIf` flag, which was failing for me before changes were made. 
 _(see: [#822](https://github.com/Azure/azure-sdk-tools/pull/822))_

# Last Upstream Rebase

Monday, July 27, 4:48pm (EDT)

# References and Related Issues 

- [Test Resources: Add Environment Variables](https://github.com/Azure/azure-sdk-tools/pull/822) (#822)
- [New-TestResources.ps1 should set AZURE_AUTHORITY_HOST environment variable](https://github.com/Azure/azure-sdk-tools/issues/757) (#757)
- [Event Hubs: AAD Auth failure when testing against other clouds](https://github.com/Azure/azure-sdk-for-net/issues/12964) (#12964)
- [Event Hub Live Test Change](https://github.com/Azure/azure-sdk-for-net/issues/13217) (#13217)

### CODE REVIEW

**Review by pakrym — APPROVED** [review]
Great change, make sure you change the shared scripts in https://github.com/Azure/azure-sdk-tools/tree/master/eng/common/TestResources first and let them sync.

File: eng/common/TestResources/New-TestResources.ps1 [hunk 1]
```diff
@@ -229,6 +235,13 @@ if ($resourceGroup.ProvisioningState -eq 'Succeeded') {
     # New-AzResourceGroup would've written an error and stopped the pipeline by default anyway.
     Write-Verbose "Successfully created resource group '$($resourceGroup.ResourceGroupName)'"
 }
+elseif (($resourceGroup -eq $null) -and (-not $PSCmdlet.ShouldProcess($resourceGroupName))) {
+    # If the -WhatIf flag was passed, there will be no resource group created.  Fake it.
+    $resourceGroup = [PSCustomObject]@{
```
Comment (pakrym) [inline eng/common/TestResources/New-TestResources.ps1:240]: ❤️

File: eng/common/TestResources/New-TestResources.ps1 [hunk 2]
```diff
@@ -154,14 +154,21 @@ if ($ProvisionerApplicationId) {
     $subscriptionArgs = if ($SubscriptionId) {
```
Comment (weshaggard) [inline eng/common/TestResources/New-TestResources.ps1:154]: Any changes under eng/common will be overwritten if they aren't made in the azure-sdk-tools repo as well, see https://github.com/Azure/azure-sdk-for-net/tree/master/eng/common#updating

Comment (jsquire) [reply eng/common/TestResources/New-TestResources.ps1:154]: Thanks for the heads-up.  I was not aware.

**Comment by jsquire** [comment]
> Great change, make sure you change the shared scripts in https://github.com/Azure/azure-sdk-tools/tree/master/eng/common/TestResources first and let them sync.

Thanks!  I wasn't aware of the mirroring.

File: eng/common/TestResources/New-TestResources.ps1 (hunk 2 above)
Comment (jsquire) [reply eng/common/TestResources/New-TestResources.ps1]: These changes have been moved to the appropriate location in the tools repo.

**Pull request merged**